lib/osf-components/addon/components/search-result-card/component.ts

```diff
@@ -1,31 +1,81 @@
 import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { waitFor } from '@ember/test-waiters';
+import Store from '@ember-data/store';
 import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
-import SearchResultModel from 'ember-osf-web/models/search-result';
-import { inject as service } from '@ember/service';
 import Intl from 'ember-intl/services/intl';
+import { dropTask } from 'ember-concurrency';
+import { taskFor } from 'ember-concurrency-ts';
+
+import config from 'ember-osf-web/config/environment';
+import SearchResultModel from 'ember-osf-web/models/search-result';
+import UserModel from 'ember-osf-web/models/user';
+
+const osfUrl = config.OSF.url;
 
 interface Args {
     result: SearchResultModel;
 }
 
 export default class SearchResultCard extends Component<Args> {
     @service intl!: Intl;
+    @service store!: Store;
+
     @tracked isOpenSecondaryMetadata = false;
+    @tracked osfUser?: UserModel;
 
     @action
     toggleSecondaryMetadata() {
         this.isOpenSecondaryMetadata = !this.isOpenSecondaryMetadata;
+        if (this.isUserResultCard && !this.osfUser) {
+            taskFor(this.getOsfUserModel).perform();
+        }
     }
 
     get cardTypeLabel() {
         return this.intl.t(`osf-components.search-result-card.${this.args.result.resourceType}`);
     }
 
-    // not sure if this is the best way, as there was a resourceType of "unknown" out in the wild
+    get isUserResultCard() {
+        return this.args.result.resourceType === 'user';
+    }
+
     get secondaryMetadataComponent() {
         const { resourceType } = this.args.result;
 
         return `search-result-card/${resourceType.replace('_component', '')}-secondary-metadata`;
     }
+
+    @dropTask
+    @waitFor
+    async getOsfUserModel() {
+        const { result } = this.args;
+        if (result.resourceType === 'user') {
+            const { identifier } = result.resourceMetadata;
+            if (identifier) {
+                const guid = this.guidFromIdentifierList(identifier);
+                if (guid) {
+                    const user = await this.store.findRecord('user', guid, {
+                        adapterOptions: {
+                            query: {
+                                related_counts: 'nodes,registrations,preprints',
+                            },
+                        },
+                        reload: true,
+                    });
+                    this.osfUser = user;
+                }
+            }
+        }
+    }
+
+    guidFromIdentifierList(ids: Array<{'@value': string}>) {
+        const osfId = ids.find(id => id['@value'].includes(osfUrl))?.['@value'];
+        if (osfId) {
+            // remove osfUrl from id and any leading/trailing slashes
+            return osfId.replace(osfUrl, '').replace(/^\/|\/$/g, '');
```

Comment:
This feels like a fragile way of fetching the user guid. Any thoughts/feedback on this would be appreciated!

Reply:
I suppose this could be done fairly simply with the URL object type by doing something like

```
for (const iri of irisArray ) {
   const url = new URL(iri);
   if (url.host === osfUrl) {
      return url.pathname.slice(1); // remove leading slash. Could also be slice(1,6)??
   }
}
```
Although this approach feels a bit too naive to make into a generalized util function

Reply:
if you want to compare hosts, yeah the built-in `URL` seems better than parsing it yourself (tho you'd also need to extract the host from `osfUrl`, since it's a full url) -- at the moment i'm more a fan of the primitive startswith/slice approach for iris within namespaces (one way to look at osf-id (osf-guid)), tho there's also the complication of osf-urls that are not osf-ids but are still used as identifiers...

```js
for (const iri of ids.map(id => id['@value'])) {
    if (iri && iri.startsWith(osfUrl)) { 
        const pathSegments = iri.slice(osfUrl.length).split('/').filter(Boolean);
        if (pathSegments.length === 1) {
            return pathSegments[0];  // one path segment; looks like osf-id
        }
    } 
}
```
or using `URL` and comparing hosts:
```js
const osfHost = new URL(osfUrl).host;
for (const iri of ids.map(id => new URL(id['@value']))) {
    if (iri.host === osfHost) { 
        const pathSegments = iri.pathname.split('/').filter(Boolean);
        if (pathSegments.length === 1) {
            return pathSegments[0];  // one path segment; looks like osf-id
        }
    } 
}
```
(the two would have different behavior if there were ever an `osfUrl` that had a non-empty path, which might be a nice future to aim for (supporting folks who want to deploy an instance at a subroute under their own domain) but is not the present we're in)

buuuut in any case, maybe the eventual helpful reusable thing would be a method on `IndexCardModel` to extract an osf-id from `this.resourceIdentifier` and/or `this.resourceMetadata.identifier` (and maybe another method to fetch the ember-data model instance for it) -- no pressure to add that now, tho

Reply:
I would not have caught the subtle difference in these two approaches. I'll opt for the first one that uses `startsWith`, as I think there may be some cases where the `pathname` approach may cause problems. Will add this guid-extration method and a `fetchOsfModel`, as I think that would be handy for possible future improvements